sympy/matrices/matrices.py

```diff
@@ -2561,10 +2561,29 @@ def diagonal_solve(self, rhs):
             raise TypeError("Size mis-match")
         return self._diagonal_solve(rhs)
 
-    def dot(self, b):
-        """Return the dot product of two vectors of equal length. ``self`` must
-        be a ``Matrix`` of size 1 x n or n x 1, and ``b`` must be either a
-        matrix of size 1 x n, n x 1, or a list/tuple of length n. A scalar is returned.
+    def dot(self, b, hermitian=None, conjugate_convention=None):
+        """Return the dot product of two vectors of equal length. ``self``
+        must be a ``Matrix`` of size 1 x n or n x 1, and ``b`` must be
+        either a matrix of size 1 x n, n x 1, or a list/tuple of length n.
+        A scalar is returned.
+
+        In case one or both of the vectors are complex and the hermitian
+        flag is set to true we need to calculate the hermitian inner
+        product of the two vectors.
+
+        The expected kwargs are hermitian and conjugate_convention.
+        Hermitian takes argument as either True or False depending on
+        whether dot product is to be calculated or hermitian inner
+        product is to be calculated.
+
+        Conjugate_convention takes one argument as input. The accepted
+        arguments are: "maths", "left", "math", "physics" or "right".
+        "maths", "math" or "left" all correspond to the same convention
+        in which hermitian of the first vector is used.
+        i.e. (a.conjugate() * b)[0]
```

Comment:
"in which the hermitian conjugate of the first vector is used."

The "i.e." part is a sentence fragment.  Should be "...is used, i.e., meh".

But also `a` is undefined, its not clear `*` DTRT here, I have no idea why `[0]` is there, and this needs to be inside dbl tics.  I do like the idea of having an example but this might do more harm than good: maybe just remove it?